autoblocks/_impl/testing/evaluators/llm_judge.py

```diff
@@ -0,0 +1,186 @@
+import abc
+import json
+from textwrap import dedent
+from typing import Any
+from typing import Generic
+from typing import List
+from typing import Optional
+
+from autoblocks._impl import global_state
+from autoblocks._impl.config.constants import API_ENDPOINT
+from autoblocks._impl.testing.evaluators.util import get_autoblocks_api_key
+from autoblocks._impl.testing.evaluators.util import get_openai_client
+from autoblocks._impl.testing.evaluators.util import get_test_id
+from autoblocks._impl.testing.models import BaseTestEvaluator
+from autoblocks._impl.testing.models import Evaluation
+from autoblocks._impl.testing.models import EvaluatorOverride
+from autoblocks._impl.testing.models import EvaluatorOverrideComment
+from autoblocks._impl.testing.models import EvaluatorOverrideField
+from autoblocks._impl.testing.models import OutputType
+from autoblocks._impl.testing.models import ScoreChoice
+from autoblocks._impl.testing.models import TestCaseType
+from autoblocks._impl.testing.models import Threshold
+
+function_name = "select_answer"
+
+
+class BaseLLMJudge(BaseTestEvaluator, abc.ABC, Generic[TestCaseType, OutputType]):
+    """
+    Base evaluator for creating an LLM judge.
+    """
+
+    @property
+    def threshold(self) -> Optional[Threshold]:
+        """
+        The threshold for the evaluator.
+        """
+        return None
+
+    @property
+    def model(self) -> str:
+        """
+        The model to use for the evaluator.
+        It must be an OpenAI model that supports tools.
+        Defaults to "gpt-4-turbo".
+        """
+        return "gpt-4-turbo"
+
+    @property
+    def no_of_overrides(self) -> int:
+        """
+        The number of recent overrides to fetch for the evaluator and pass to make_prompt.
+
+        Defaults to 0.
+        """
+        return 0
+
+    @abc.abstractmethod
+    def make_prompt(
+        self, test_case: TestCaseType, output: OutputType, recent_overrides: List[EvaluatorOverride]
+    ) -> str:
+        """
+        The prompt passed to the LLM judge. Should be poised as a question.
+        """
+        pass
+
+    @property
+    @abc.abstractmethod
+    def score_choices(self) -> List[ScoreChoice]:
+        """
+        The choices for the LLM judge to use when answering.
+        """
+        pass
+
+    def _make_score_choice(self, value: float) -> Optional[ScoreChoice]:
+        return next((score for score in self.score_choices if score.value == value), None)
```

Comment:
could you add a comment on why you're using `next` here?

Reply:
renamed the function and added a comment so its more clear

Reply:
ah yeah i meant specifically why `next` is being used, haven't really seen it used in this way. it seems you could do:

```python
for score in self.score_choices:
  if score.value == value:
    return score
return None
```

Reply:
I got it from https://stackoverflow.com/a/9542768